### PR TITLE
Merge KittenPopo Additions

### PR DESCRIPTION
**Pull request by KittenPopo**
Seems pretty solid, tested it until I got bored.

Change summary:
- Fixed grapple ExploitFixes "error log" bug
- Added advanced debug info dump on crash to allow debugging of stack-corrupted crashes
- Improved KHook/NSMem stuff
- Default value for net_chan_limit_msec_per_sec to avoid dumb console spam
- Added debug-only logging
- Disable auth kicks when playing local games
- Added ANTI-RCE system (basically just damage control for now, prevent file system access to places titanfall should never be accessing)
- Fix for a bunch of console command exploits, including the infamous instant-respawn exploit (I also merged emit_blocker with ExploitFixes.cpp as part of the hook I made)

### CODE REVIEW

**Comment by KittenPopo** [comment]
**NOTE**: NSQApi stuff is _NOT_ present in this pull request - don't worry about that.

File: NorthstarDedicatedTest/NorthstarDedicatedTest.vcxproj [hunk 1]
```diff
@@ -117,7 +117,6 @@
     <ClInclude Include="debugoverlay.h" />
     <ClInclude Include="clientruihooks.h" />
     <ClInclude Include="clientvideooverrides.h" />
-    <ClInclude Include="emit_blocker.h" />
```
Comment (GeckoEidechse) [inline NorthstarDedicatedTest/NorthstarDedicatedTest.vcxproj:120]: Uh, probably don't wanna remove that :P

**Review by GeckoEidechse — DISMISSED** [review]

File: NorthstarDedicatedTest/NorthstarDedicatedTest.vcxproj (hunk 1 above)
Comment (GeckoEidechse) [reply NorthstarDedicatedTest/NorthstarDedicatedTest.vcxproj:120]: Nevermind, got moved into `NorthstarDedicatedTest/ExploitFixes.cpp

**Comment by KittenPopo** [comment]
Added exploit fixes, "consumed" emit_blocker hotfix into ExploitFixes.cpp hook.

image

**Comment by GeckoEidechse** [comment]
> I am also considering adding https://github.com/KittenPopo/NorthstarLauncher/tree/no-more-spdlog to this pr... :flushed:

That should be it's own PR if you want it changed. General rule of thumb: _1 PR == 1 feature change_

**Comment by KittenPopo** [comment]
> > I am also considering adding https://github.com/KittenPopo/NorthstarLauncher/tree/no-more-spdlog to this pr... 😳
> 
> That should be it's own PR if you want it changed. General rule of thumb: _1 PR == 1 feature change_

Fair enough, that's gonna be a lot of PRs xD

**Review by EladNLG — CHANGES_REQUESTED** [review]

File: NorthstarDedicatedTest/clientchathooks.cpp [hunk 2]
```diff
@@ -54,42 +54,8 @@ static void CHudChat__AddGameLineHook(void* self, const char* message, int inbox
...
-	LocalChatWriter((LocalChatWriter::Context)context).WriteLine(str);
-	return SQRESULT_NOTNULL;
-}
-
 void InitialiseClientChatHooks(HMODULE baseAddress)
 {
 	HookEnabler hook;
 	ENABLER_CREATEHOOK(hook, (char*)baseAddress + 0x22E580, &CHudChat__AddGameLineHook, reinterpret_cast<LPVOID*>(&CHudChat__AddGameLine));
-
-	g_ClientSquirrelManager->AddFuncRegistration("void", "NSChatWrite", "int context, string text", "", SQ_ChatWrite);
```
[the remark this answers is not in the document]
Comment (KittenPopo) [reply NorthstarDedicatedTest/clientchathooks.cpp:92]: Fixed, thx. I was implementing a much cleaner and more format squirrel extension system that I removed in this PR, forgot to restore original functions (?).

[the remark this answers is not in the document]
Comment (EladNLG) [reply NorthstarDedicatedTest/clientchathooks.cpp:92]: Restored, absolutely pog

**Review by GeckoEidechse — COMMENTED** [review]
Regarding all the changes from `spdlog::info` to `DBLOG`, what exactly is the reason for it? ^^"

**Comment by KittenPopo** [comment]
> Regarding all the changes from `spdlog::info` to `DBLOG`, what exactly is the reason for it? ^^"

Such that spammy log messages don't fill the console except in debug mode.
In my no-more-spdlog pr it will log to a file but not print in console unless _DEBUG.

**Comment by GeckoEidechse** [comment]
> Such that spammy log messages don't fill the console except in debug mode.

I mean, just use `spdlog::debug` and set log level appropriately (not sure if we do the second part yet).

**Comment by Mauler125** [comment]
> 

Why not guard the cout behind a ConVar? Also for spdlog you could set the level globally and exclude everything all together with 1 line. I was working on implementing a fast packet dumper with spdlog but the replacement with this new logger will break my PR.

**Review by EladNLG — APPROVED** [review]

**Comment by KittenPopo** [comment]
> > 
> 
> Why not guard the cout behind a ConVar? Also for spdlog you could set the level globally and exclude everything all together with 1 line. I was working on implementing a fast packet dumper with spdlog (to dump the decrypted payload) but the replacement with this new logger will break my PR.

Yeah that's fair, tbh I'm just trying to get this PR done so that these fixes can get released. I told Bob this, but, feel free to revert the DLOGs back to spdlog::info.

**Comment by KittenPopo** [comment]
Splitting this up into multiple because my pull request was too based

**Pull request closed without merging**